lib/AST/Decl.cpp

```diff
@@ -11701,51 +11705,29 @@ ActorIsolation ActorIsolation::forActorInstanceSelf(NominalTypeDecl *selfDecl) {
 }
 
 NominalTypeDecl *ActorIsolation::getActor() const {
-  assert(getKind() == ActorInstance ||
-         getKind() == GlobalActor);
+  assert(getKind() == ActorInstance || getKind() == GlobalActor);
 
   if (silParsed)
     return nullptr;
 
-  Type actorType;
-
-  if (auto *instance = actorInstance.dyn_cast<VarDecl *>()) {
-    actorType = instance->getTypeInContext();
-  } else if (auto *instance = actorInstance.dyn_cast<Expr *>()) {
-    actorType = instance->getType();
-  }
-
-  if (actorType) {
-    if (auto wrapped = actorType->getOptionalObjectType()) {
-      actorType = wrapped;
-    }
-    return actorType
-        ->getReferenceStorageReferent()->getAnyActor();
+  if (getKind() == GlobalActor) {
+    return getGlobalActor()->getAnyNominal();
```

Comment:
`ActorIsolation::getActor()` used to return garbage before this fix.

We initialize the `globalActor` property but fall through to returning the `actorInstance.get<NominalTypeDecl *>();` which is just some uninitialized memory for the global actor case.

This is fixed now, we properly handle both kinds of isolations the method claimed to support.

Just FYI @hborla